master/security/kubernetes-node-ports.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Apply policy to Kubernetes node ports
+---
+
+### Big picture
+
+Restrict access to node ports to specific external clients.
+
+### Value
+
+Exposing services to external clients using node ports is a standard Kubernetes feature. However, if you want to restrict access to node ports to specific external clients, you need to use Calico global network policy.
+
+### Features
+
+This how-to guide uses the following Calico features:
+- **GlobalNetworkPolicy** with a preDNAT field
+- **HostEndpoint**
+
+### Concepts
+
+#### Calico global network policy with a preDNAT field 
+
+In a Kubernetes cluster, kube-proxy will DNAT a request to the node's port and IP address, to one of the pods that backs the service. For Calico global network policy to both allow normal ingress cluster traffic, and deny other general ingress traffic, it must take effect before DNAT. To do this, you simply add a **preDNAT** field to a Calico global network policy. The preDNAT field:
+
+- Applies only to ingress rules
+- Enforces all ingress traffic through a host endpoint, regardless of destination.  
+  The destination can be a locally hosted pod, a pod on another node, or a process running on the host. 
+
+
+### Before you begin...
+
+- For services that you want to expose to external clients, configure Kubernetes services with type **NodePort**. 
+- Define Calico [host endpoints]({{site.baseurl}}/{{page.version}}/reference/resources/hostendpoint) for the Kubernetes nodes with network policy that is appropriate for your cluster.
+
+
+### How to
+
+To securely expose a Kubernetes service to external clients, you must implement all of the following steps. 
+
+- [Allow cluster ingress traffic, but deny general ingress traffic](#allow-cluster-ingress-traffic-but-deny-general-ingress-traffic)
+- [Allow local host egress traffic](#allow-local-host-egress-traffic)
+- [Create host endpoints with appropriate network policy](#create-host-endpoints-with-appropriate-network-policy)
```

Comment:
I think this section overlaps so heavily with the before you begin prerequisite that people might skip it (it's the exact same heading).  I'd be tempted to only have the section below and incorporate the link out.  It's not like there's anywhere in the site that helps people with the actually hard part (defining appropriate policy that allows the node to still function).

Reply:
Fixed.